File: frontend/hrms/hrms/src/services/employerRecourseService.js
import axios from "axios"

export default class EmployerRecourseService{
    add(values){
        return axios.post("http://localhost:8080/api/employerrecourse/add",values)
    }

    delete(id){
        return axios.delete(`http://localhost:8080/api/employerrecourse/deletebyid?id=${id}`)
    }

    update(recourse){
        return axios.put(`http://localhost:8080/api/employerrecourse/update`,recourse)
    }

    getByEmployerId(id){
        return axios.get(`http://localhost:8080/api/employerrecourse/getbyemployerid?id=${id}`)
    }
    
    getByCandidateId(id){
        return axios.get(`http://localhost:8080/api/employerrecourse/getbycandidateid?id=${id}`)
    }

    getByAdvertId(id){
        return axios.get(`http://localhost:8080/api/employerrecourse/getbyadvertid?id=${id}`)
    }
}

File: backend/Odevler/Gun_7/Odev_1/hrms/src/main/java/javaReact_BootCamp/hrms/business/abstracts/EmployerRecourseService.java
package javaReact_BootCamp.hrms.business.abstracts;

import java.util.List;

import javaReact_BootCamp.hrms.core.utilites.result.DataResult;
import javaReact_BootCamp.hrms.core.utilites.result.Result;
import javaReact_BootCamp.hrms.entities.concretes.EmployerRecourse;

public interface EmployerRecourseService {
	DataResult<List<EmployerRecourse>> getAll();
	
	DataResult<List<EmployerRecourse>> getByCandidateId(int id);
	
	DataResult<List<EmployerRecourse>> getByEmployerId(Integer id);
	
	DataResult<List<EmployerRecourse>> getByAdvertId(Integer id);
	
	DataResult<List<EmployerRecourse>> getByStatus(String status);
	
	DataResult<EmployerRecourse> getByEmployerIdAndAdvertId(Integer employerId,Integer advertId );
	
	Result add(EmployerRecourse recourse);
	
	Result delete(int id);
	
	Result update(EmployerRecourse recourse);
}
